src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -541,8 +541,7 @@ private void CompileNamedType(NamedTypeSymbol containingType)
 
                     case SymbolKind.Field:
                         {
-                            var field = (FieldSymbol)member;
-                            var fieldSymbol = (field.TupleUnderlyingField ?? field) as SourceMemberFieldSymbol;
+                            FieldSymbol fieldSymbol = member as FieldSymbol;
                             if ((object)fieldSymbol != null)
```

Comment:
Can use explicit cast and avoid the `null` check.

Reply:
Could also do 

```cs
if (member is FieldSymbol fieldSymbol)
```